praetorian_cli/ui/console/commands/context.py

```
@@ -14,11 +14,28 @@ def _cmd_set(self, args):
         if len(args) < 2:
             self.console.print('[dim]Usage: set <account|scope|mode|target> <value>[/dim]')
             return
-        key, value = args[0].lower(), ' '.join(args[1:])
+        key, value = args[0].lower(), ' '.join(args[1:]).strip().rstrip('.')
         if key == 'account':
+            # Always validate against the real account list
+            if not (hasattr(self, '_account_list') and self._account_list):
+                try:
+                    accounts_list, _ = self.sdk.accounts.list()
+                    self._account_list = [a.get('name', '') for a in (accounts_list or [])]
+                except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)